scripts/lint

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+set -euf -o pipefail
+
+node_modules/.bin/sanctuary-lint "$@"
+
+function rewrite {
+  local state=
+  local line
+  while IFS='' read -r line ; do
+    local a="${line:0:1}"
+    local b="${line:1:1}"
+    local z="${line: -1}"
+    if [[ $a$b == '> ' ]] ; then
+      printf '\n%s' "${line:2}"
+      state=input
+    elif [[ -n $state && $a$b == '. ' ]] ; then
+      printf '\n%s' "${line:2}"
+    elif [[ $state == input && $a$z == '{}' ]] ; then
+      printf ';\n%s' "($line)"
+      state=output
+    elif [[ $state == input ]] ; then
+      printf ';\n%s' "$line"
+      state=output
+    elif [[ $state == output ]] ; then
+      printf ';\n%s' "$line"
+      state=
+    else
+      printf '\n%s' "$line"
+    fi
+  done
+}
+
+trap 'mv README.md.orig README.md && rm -f README.md.temp' EXIT
+cp README.md README.md.orig
+VERSION=0.0.0 PREVIOUS_VERSION=0.0.0 node_modules/.bin/sanctuary-generate-readme
+cp README.md README.md.temp
+rewrite <README.md.temp >README.md
+
+node_modules/.bin/eslint --config eslint/es6.js -- README.md eslint
```

Comment:
This script enables `npm run lint` to run `eslint` on files that would not otherwise be linted.

Linting the readme involves a number of steps:

1.  Save a copy of the file as it exists.
2.  Regenerate the readme (the `VERSION` and `PREVIOUS_VERSION` values are unimportant).
3.  Use `rewrite` to remove leading `>` and `.` characters, parenthesize object literals which would otherwise be parsed as blocks, and insert a semicolon at the end of each input and output line.
4.  Run `eslint`.
5.  Restore the readme to its original state (via an EXIT trap).

Reply:
sanctuary-js/sanctuary-scripts#13